File: src/cosmic_pi_network/cvlib/image.py
import cv2
import numpy as np

class Image:
    def __init__(self, path: str):
        self.path = path
        self.image = cv2.imread(path)

    def resize(self, width: int, height: int) -> None:
        self.image = cv2.resize(self.image, (width, height))

    def convert_to_grayscale(self) -> None:
        self.image = cv2.cvtColor(self.image, cv2.COLOR_BGR2GRAY)

    def get_image(self) -> np.ndarray:
        return self.image

File: src/cosmic_pi_network/cvlib/cpp/Image.cpp
#include "Image.h"

Image::Image(const std::string& path) : path_(path) {
    image_ = cv::imread(path_);
}

void Image::resize(int width, int height) {
    image_ = cv::resize(image_, cv::Size(width, height));
}

void Image::convert_to_grayscale() {
    image_ = cv::cvtColor(image_, cv::COLOR_BGR2GRAY);
}

cv::Mat Image::get_image() {
    return image_;
}
